File: src/fabric/common/main/java/settingdust/mod_sets/fabric/util/LoaderAdapter.java
package settingdust.mod_sets.fabric.util;

import net.fabricmc.api.EnvType;
import net.fabricmc.loader.api.FabricLoader;
import net.fabricmc.loader.impl.FabricLoaderImpl;
import settingdust.mod_sets.ModSets;

import java.nio.file.Path;

public class LoaderAdapter implements settingdust.mod_sets.util.LoaderAdapter {
    private final Path configDirectory = FabricLoader.getInstance().getConfigDir().resolve(ModSets.ID);
    private final Path modsDirectory = FabricLoaderImpl.INSTANCE.getModsDirectory().toPath();

    public static boolean inGame = false;

    @Override
    public boolean isClient() {
        return FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT;
    }

    @Override
    public boolean isModLoaded(final String modId) {
        return FabricLoader.getInstance().isModLoaded(modId);
    }

    @Override
    public Path getConfigDirectory() {
        return configDirectory;
    }

    @Override
    public Path getModsDirectory() {
        return modsDirectory;
    }

    @Override
    public boolean isInGame() {
        return inGame;
    }
}

File: src/fabric/common/main/kotlin/settingdust/mod_sets/fabric/util/LoaderAdapter.kt
package settingdust.mod_sets.fabric.util

import net.fabricmc.api.EnvType
import net.fabricmc.loader.api.FabricLoader
import net.fabricmc.loader.impl.FabricLoaderImpl
import settingdust.mod_sets.ModSets
import settingdust.mod_sets.util.LoaderAdapter
import java.nio.file.Path
import kotlin.io.path.div

class LoaderAdapter : LoaderAdapter {
    override val isClient = FabricLoader.getInstance().environmentType === EnvType.CLIENT

    override fun isModLoaded(modId: String) = FabricLoader.getInstance().isModLoaded(modId)

    override val configPath by lazy { FabricLoader.getInstance().configDir / ModSets.ID }

    override val modsDir: Path = FabricLoaderImpl.INSTANCE.modsDirectory.toPath()
}
